x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -83,13 +88,18 @@
     private final Client client;
     private final CcrLicenseChecker ccrLicenseChecker;
 
+    private final RateLimiter.SimpleRateLimiter rateLimiter;
+    private final CounterMetric throttledTime = new CounterMetric();
+    private final AtomicLong bytesSinceLastPause = new AtomicLong();
+
     public CcrRepository(RepositoryMetaData metadata, Client client, CcrLicenseChecker ccrLicenseChecker, Settings settings) {
         super(settings);
         this.metadata = metadata;
         assert metadata.name().startsWith(NAME_PREFIX) : "CcrRepository metadata.name() must start with: " + NAME_PREFIX;
         this.remoteClusterAlias = Strings.split(metadata.name(), NAME_PREFIX)[1];
         this.ccrLicenseChecker = ccrLicenseChecker;
         this.client = client;
+        this.rateLimiter = new RateLimiter.SimpleRateLimiter(FOLLOWER_RECOVERY_MAX_BYTES_READ_PER_SECOND.get(settings).getMbFrac());
```

Comment:
This setting is dynamic, so let's allow to dynamically adjust it by registering a settings update consumer (see the `RecoverySettings` how it's done). We can use the CCRSettings object to store the latest value, similar to the `RecoverySettings` object. You can create the `CCSettings` instance in the `CCR` class and do the registration with CLusterSettings in createComponents, which can be accessed from clusterService.getClusterSettings()

Reply:
I registered a listener for the setting change.